apps/meteor/ee/server/models/raw/LivechatRooms.ts

```diff
@@ -10,14 +10,93 @@ declare module '@rocket.chat/model-typings' {
 	export interface ILivechatRoomsModel {
 		associateRoomsWithDepartmentToUnit: (departments: string[], unit: string) => Promise<void>;
 		removeUnitAssociationFromRooms: (unit: string) => Promise<void>;
-		updateDepartmentAncestorsById: (unitId: string, ancestors: string[]) => Promise<void>;
+		updateDepartmentAncestorsById: (rid: string, ancestors: string[]) => Promise<void>;
 		unsetPredictedVisitorAbandonmentByRoomId(rid: string): Promise<UpdateResult>;
-		findAbandonedOpenRooms(date: Date): Promise<FindCursor<IOmnichannelRoom>>;
+		findAbandonedOpenRooms(date: Date): FindCursor<IOmnichannelRoom>;
+		setPredictedVisitorAbandonmentByRoomId(roomId: string, date: Date): Promise<UpdateResult>;
+		unsetAllPredictedVisitorAbandonment(): Promise<void>;
+		setOnHoldByRoomId(roomId: string): Promise<UpdateResult>;
+		unsetOnHoldByRoomId(roomId: string): Promise<UpdateResult>;
+		unsetOnHoldAndPredictedVisitorAbandonmentByRoomId(roomId: string): Promise<UpdateResult>;
+		unsetPriorityByIdFromAllOpenRooms(priorityId: string): Promise<void>;
+		findOpenRoomsByPriorityId(priorityId: string): FindCursor<IOmnichannelRoom>;
 	}
 }
 
+// Note: Expect a circular dependency error here 😓
 export class LivechatRoomsRawEE extends LivechatRoomsRaw implements ILivechatRoomsModel {
-	async findAbandonedOpenRooms(date: Date): Promise<FindCursor<IOmnichannelRoom>> {
+	async unsetAllPredictedVisitorAbandonment(): Promise<void> {
+		await this.updateMany(
```

Comment:
Do you really need to await here? Can't the parent function await this one?

Reply:
Indeed they can :smile: But I think I earlier added an await here because the return type of this function wasn't the same as the return type of updateMany. 

On second thought, I think I can modify the return type within a promise handler so we can avoid await here